cspotcode/node-source-map-support
# Tests, improved ESM support, switch to @jridgewell/trace-mapping, etc

This ended up being a jumbo PR doing lots of improvements for the next release.

Implements and closes #37

https://github.com/TypeStrong/ts-node/issues/1726
https://github.com/TypeStrong/ts-node/pull/1729

Also adds tests for ESM and for ts-node's style of sourcemap.

Full list of changes:

- switch to @jridgewell/trace-mapping; away from fork of source-map-consumer
  - goal is to use lib that is maintained by someone else, and reap improved performance
- add node 17 and 18 to test matrix
- normalize cache keys: URI and native path shared cache entry
- add tests for ESM (previously was only CJS)
- add tests for new path styles in sourcemaps: absolute paths, file URIs
- ensure formatting of stack frames and path above snippet is consistent with generated JS
  - stack frame/snippet from nodejs may be native path or file URI
    - CJS uses native paths
    - ESM uses file URIs
  - mapped frame should match: either native path or file URI
  - Note: On windows, this means backslashes, starts with drive letter, no file:// prefix
- add test for async stack frames
  - async, Promise.all, Promise.any
- add support for wasm stack frames
- Fixed bug where file:/// URLs in stack traces were not consistently percent-encoded
- re-add browser-source-map-support.js so that legacy stuff which tries to get it from us will succeed
